crates/fuel-core/src/database/vm_database.rs

```diff
@@ -227,47 +227,16 @@ impl InterpreterStorage for VmDatabase {
         start_key: &Bytes32,
         range: usize,
     ) -> Result<Vec<Option<Cow<Bytes32>>>, Self::DataError> {
-        // TODO: Optimization: Iterate only over `range` elements.
-        let mut iterator = self.database.iter_all_filtered::<Vec<u8>, Bytes32, _, _>(
-            Column::ContractsState,
-            Some(contract_id),
-            Some(ContractsStateKey::new(contract_id, start_key)),
-            Some(IterDirection::Forward),
-        );
-
-        let mut expected_key = U256::from_big_endian(start_key.as_ref());
-        let mut results = vec![];
-
-        while results.len() < range {
-            let entry = iterator.next().transpose()?;
-
-            if entry.is_none() {
-                // We out of `contract_id` prefix
-                break
-            }
+        let mut multikey = [0u8; 64];
+        multikey[..32].copy_from_slice(contract_id.as_ref());
+        let mut key = U256::from_big_endian(start_key.as_ref());
 
-            let (multikey, value) =
-                entry.expect("We did a check before, so the entry should be `Some`");
-            let actual_key = U256::from_big_endian(&multikey[32..]);
-
-            while (expected_key <= actual_key) && results.len() < range {
-                if expected_key == actual_key {
-                    // We found expected key, put value into results
-                    results.push(Some(Cow::Owned(value)));
-                } else {
-                    // Iterator moved beyond next expected key, push none until we find the key
-                    results.push(None);
-                }
-                expected_key.increase()?;
-            }
-        }
-
-        // Fill not initialized slots with `None`.
-        while results.len() < range {
-            results.push(None);
-            expected_key.increase()?;
+        let mut results = Vec::new();
+        for _ in 0..range {
+            key.to_big_endian(&mut multikey[32..]);
+            results.push(self.database.get(&multikey, Column::ContractsState)?);
```

Comment:
Please use `storage::<ContractsState>().get` instead. In this case we can extract `VmDatabase` from `fuel-core`(we don't need to know about `Column`).